src/map/mob.cpp

```diff
@@ -1871,12 +1919,10 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 	{
 		int32 prev_id = md->target_id;
 		map_foreachinallrange (mob_ai_sub_hard_activesearch, &md->bl, view_range, DEFAULT_ENEMY_TYPE(md), md, &tbl, mode);
-		// If a monster finds a target through search that is already in attack range it immediately switches to berserk mode
+		// If a monster finds a target through search that is already in attack range it switches to berserk mode
 		// This behavior overrides even angry mode and other mode-specific behavior
-		if (tbl != nullptr && prev_id != md->target_id && battle_check_range(&md->bl, tbl, md->status.rhw.range)) {
+		if (tbl != nullptr && prev_id != md->target_id && battle_check_range(&md->bl, tbl, md->status.rhw.range))
 			md->state.aggressive = 0;
-			md->state.skillstate = MSS_BERSERK;
```

Comment:
Comment still says it switches mode. Did you intentionally remove this or should we call mob_setstate here?

Reply:
I removed the word "immediately" because it doesn't set the mode here but rather later in the code (after the attack).

It's related to this from PR description:

- A monster that finds an enemy in attack range will now start off with a normal attack and then switch to Berserk
  * This still overwrites angry mode as before

The confusing part is a bit because we sometimes call setting "aggressive = 0" as "leaving angry mode / entering berserk mode". Maybe I need to elaborate it in the comment a little.

Reply:
Comment improved.